Gordon360/Services/MembershipRequestService.cs

```diff
@@ -113,150 +75,86 @@ public MEMBERSHIP ApproveRequest(int requestID)
         /// </summary>
         /// <param name="requestID">The membership request id</param>
         /// <returns>A copy of the deleted membership request</returns>
-        public REQUEST Delete(int requestID)
+        public async Task<RequestView> DeleteAsync(int requestID)
         {
-            var result = _context.REQUEST.Find(requestID);
-            if (result == null)
+            var request = await _context.REQUEST.FindAsync(requestID);
+            if (request == null)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The Request was not found." };
             }
-            _context.REQUEST.Remove(result);
-            _context.SaveChanges();
-            return result;
+
+            RequestView rv = _context.RequestView.FirstOrDefault(r => r.RequestID == request.REQUEST_ID);
+
+            _context.REQUEST.Remove(request);
+            await _context.SaveChangesAsync();
+
+            return rv;
         }
 
         /// <summary>
         /// Denies the membership request object whose id is given in the parameters
         /// </summary>
         /// <param name="requestID">The membership request id</param>
         /// <returns></returns>
-        public REQUEST DenyRequest(int requestID)
+        public async Task<RequestView> DenyAsync(int requestID)
         {
-            var query = _context.REQUEST.Find(requestID);
-            if (query == null)
+            var request = await _context.REQUEST.FindAsync(requestID);
+
+            if (request == null)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The Request was not found." };
             }
 
-            query.STATUS = Request_Status.DENIED;
-            _context.SaveChanges();
-            return query;
+            request.STATUS = Request_Status.DENIED;
+            await _context.SaveChangesAsync();
+
+            return _context.RequestView.FirstOrDefault(r => r.RequestID == request.REQUEST_ID);
         }
 
         /// <summary>
         /// Get the membership request object whose Id is specified in the parameters.
         /// </summary>
         /// <param name="requestID">The membership request id</param>
         /// <returns>If found, returns MembershipRequestViewModel. If not found, returns null.</returns>
-        public async Task<MembershipRequestViewModel> GetAsync(int requestID)
+        public RequestView Get(int requestID)
         {
-            var requests = await _context.Procedures.REQUEST_PER_REQUEST_IDAsync(requestID);
+            var request = _context.RequestView.FirstOrDefault(r => r.RequestID == requestID);
 
-            if (requests == null || requests.Count != 1)
+            if (request == null)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The Request was not found." };
             }
```

Comment:
Would it be better to return a nullable `RequestView?` here, instead of throwing a `ResourceNotFoundException`?

Reply:
While it does make intuitive sense to send a not found response when requesting a specific resource, I wonder if that's actually best practice.

If we think of the origins of 404, when HTTP was used to fetch specific files from a server by name, sending an error code when requesting a file that the server didn't have made perfect sense - and was a good user experience.

I'm not convinced that the same thing is true of a REST API though. Although a 'not found' response sounds right, I think that sending an error code is not really the most straight forward thing to do - because the client is not a user but an application. A user would have no idea what to if e.g. `GET /api/requests/:requestId` returned `null`. An application, on the other hand, can understand and handle `null` much more easily than `Error: 404 Not Found`. Adding an extra `catch` block to every frontend HTTP request that handles 404 is more difficult and less intuitive than handling a possible `null` response - at least in my opinion.

Consider, also, that this allows us to differentiate null responses from more meaningful 404s, such as a `404 User Not Found` when calling `GET /api/accounts/:username/memberships`. In that case, `null` simply means that the user doesn't have any memberships in the database